### ISSUE TITLE
improve systemd sandboxing and autostart services

### ISSUE THREAD

**Issue by RoyalOughtness** [secureblue/secureblue#114]
/usr/bin/systemd-analyze security SERVICE

**Comment by RoyalOughtness** [comment]
closing as on second thought, this should be handled and reported upstream.

**Comment by RoyalOughtness** [comment]
Reopening this as there is lots of room for improvement and upstream progress has been too gradual.

**Comment by ghost** [comment]
I have begun work on this, I will commit any changes I make to a testing branch [here](https://github.com/lacklustrelife/secureblue/tree/systemd-hardening). I've started with `NetworkManager` already and have a config [here](https://github.com/lacklustrelife/secureblue/blob/systemd-hardening/files/system/etc/systemd/system/NetworkManager.service.d/override.conf) that I have done basic testing with.

I am implementing the hardening configuration for each service in override files in `/etc/systemd/system/servicename.service.d/override.conf`. Configuration made in these files overrides default service configurations located at `/usr/lib/systemd/servicename.service`.

I've create a thread in the discord for this task as it is going to take a lot of trial and error. If anyone else wants to start hardening some services, leave a comment on this issue or in the [thread](https://discord.com/channels/1202086019298500629/1420409026004385842) and we can figure out who's working on what service.

**Comment by alexvojproc** [comment]
dumping this here: https://s1m.fr/systemd-sandbox/

tldr is that these sandboxes are trivially escaped with a transient unit if you don't follow best practices:

- ideally use `DynamicUser=yes`
- **or** use a non-root user **and** `ProtectHome=true`
- **or** use a non-root user at least use `InaccessiblePaths=/run/user/<UID>/bus`
- **or**, if you must be root, use `InaccessiblePaths=/run/dbus/ /run/user/` at a minimum
  - relevant RFE: https://github.com/systemd/systemd/issues/8846

**Comment by nihil-admirari** [comment]
Consider switching to [`chronyd-restricted.service`](https://github.com/mlichvar/chrony/blob/master/examples/chronyd-restricted.service) (ships with Fedora [by default, with a special SELinux profile](https://src.fedoraproject.org/rpms/chrony/blob/rawhide/f/chrony.spec#_98)).

```ini
# This is a more restricted version of the chronyd service intended for
# minimal NTP/NTS client configurations. The daemon is started without root
# privileges and is allowed to write only to its own runtime, state, and log
# directories. It cannot bind to privileged ports in order to operate as an
# NTP server, or provide monitoring access over IPv4/IPv6. It cannot use
# reference clocks, HW timestamping, RTC tracking, and other features.
```

secureblue's chrony configuration references RTC (`rtcsync` and `rtconutc`), but chrony apparently never writes to RTC unless it is selected as a time source. Both `chronyc rtcdata` and `chronyc trimrtc` return `513 RTC driver not running` even on plain Fedora. [Code search](https://github.com/search?q=repo%3Amlichvar%2Fchrony+rtcsync&type=code) shows that on Linux, `rtcsync` is used to show error messages on misconfiguration and to adjust `adjtimex()` arguments, but chrony itself never writes to RTC. Arch Linux wiki [says](https://wiki.archlinux.org/title/System_time):

> By default, Arch Linux kernels have a feature enabled where the hardware clock is synchronized to the system clock every 11 minutes. You can see if this is enabled on your kernel as follows:
>
> ```console
> $ zgrep CMOS /proc/config.gz
> CONFIG_GENERIC_CMOS_UPDATE=y
> CONFIG_RTC_DRV_CMOS=y
> ```

which is further indication that RTC synchronisation is not chrony's job, but the kerne's. chrony does read RTC when [it is started with `-s`](https://github.com/mlichvar/chrony/blob/master/main.c#L588) (see https://github.com/secureblue/secureblue/issues/268#issuecomment-5158784679 for why anyone may ever use that flag).

In the light of the above, `chronyd-restricted.service` should probably be relaxed by placing

```ini
[Service]
PrivateDevices=no
DeviceAllow=char-rtc r
```

into `/etc/systemd/system/chronyd-restricted.service.d/allow-rtc-read.conf`.

After this adjustment `chronyd-restricted.service` runs with no issues

```sh
systemctl disable --now chronyd.service
systemctl mask chronyd.service
systemctl enable --now chronyd-restricted.service
```

GrapheneOS infrastructure doesn't use `chronyd-restricted.service`, apparently because [it isn't packaged by Arch](https://gitlab.archlinux.org/archlinux/packaging/packages/chrony/-/blob/main/PKGBUILD?ref_type=heads). They do [tighten the screws](https://github.com/GrapheneOS/infrastructure/blob/main/etc/systemd/system/chronyd.service.d/override.conf) loosened for [`sendmail` sake](https://github.com/mlichvar/chrony/blob/master/examples/chronyd.service#L43):

```ini
# Adjust restrictions for /usr/sbin/sendmail (mailonchange directive)
NoNewPrivileges=no
ReadWritePaths=-/var/spool
RestrictAddressFamilies=AF_NETLINK
```
